### PR TITLE
Stabilization: setup queue and handler service

### PR DESCRIPTION
**Pull request by cpgo**
Signed-off-by: Cassio Godinho <cassiopgodinho@gmail.com>

### CODE REVIEW

File: butler/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "start:prod": "node dist/main.js",
     "lint": "eslint --max-warnings 0 . --ext .js,.jsx,.ts,.tsx",
     "test": "jest --testRegex \"\\.spec\\.ts$\" ",
-    "test:integration": "jest --testRegex \"integration-spec\\.ts$\" --runInBand",
+    "test:integration": "jest --testRegex \"integration-spec\\.ts$\" --runInBand --detectOpenHandles --forceExit",
```
Comment (lucasbfernandes) [inline butler/package.json:16]: Maybe we can just force exit here. Detect open handles apparently consumes a lot of extra time sometimes

File: butler/src/app/v2/api/deployments/jobs/pgboss.worker.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
+import { Configuration } from '../../../../v1/core/config/configurations';
+import { ConsoleLoggerService } from '../../../../v1/core/logs/console';
+import { DeploymentEntity } from '../entity/deployment.entity';
+import PgBoss = require('pg-boss');
+import { DeploymentHandler } from '../use-cases/deployment-handler';
+
+@Injectable()
+export class PgBossWorker implements OnModuleInit, OnModuleDestroy{
+  pgBoss: PgBoss
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/jobs/pgboss.worker.ts:10]: I think it is better to be explicit on the visibility of the variables. What do you think? This guy is probably private

File: butler/src/app/v2/api/deployments/jobs/pgboss.worker.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  constructor(
+    private readonly consoleLoggerService: ConsoleLoggerService,
+    private readonly deploymentHandler: DeploymentHandler
+  ) {
+    const pgBossOptions = {
+      host: Configuration.postgresHost,
+      database: Configuration.postgresDbName,
+      user: Configuration.postgresUser,
+      password: Configuration.postgresPass,
+      max: 5,
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/jobs/pgboss.worker.ts:20]: Maybe create constants for this?

File: butler/src/app/v2/api/deployments/use-cases/deployment-handler.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+
+export class DeploymentHandler {
+  constructor(
+    private readonly consoleLoggerService: ConsoleLoggerService //not working but im not fighting with nest IOC now
+  ) {  }
+
+  async run(job: JobWithDoneCallback<unknown, unknown>): Promise<JobWithDoneCallback<unknown, unknown>>{
+    // this.consoleLoggerService.log('Running job', { job: job })
+    // eslint-disable-next-line no-constant-condition
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/use-cases/deployment-handler.ts:12]: Remove comments

File: butler/src/app/v2/api/deployments/use-cases/deployment-handler.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const cdServiceResponse = true
+    if (cdServiceResponse) {
+      console.log('Finished job', {job: job.id, data: job.data})
```
Comment (lucasbfernandes) [inline butler/src/app/v2/api/deployments/use-cases/deployment-handler.ts:15]: Don't console.log

Comment (cpgo) [reply butler/src/app/v2/api/deployments/use-cases/deployment-handler.ts:15]: I need to fix the ioc here, the `consoleLoggerService` is undefined.
Using console.log just for debugging the job

**Pull request merged**